Sources/StreamChat/StateLayer/UserSearch.swift

```diff
@@ -0,0 +1,67 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// An object which represents a list of `ChatUser` for the specified search query.
+@available(iOS 13.0, *)
+public struct UserSearch {
```

Comment:
It is pretty much the same as `UserList` with the two exceptions:
- `UserList` has static query, whereas here the query can change
- `UserList` is backed by CoreData observing, here we skip CoreData

Reply:
Let's keep this PR open until the cancellation part is also properly implemented.

Reply:
Don't think this PR should wait on the cancellation part. We need to handle that after we merge the open api stuff, otherwise we might need to redo it again.